modules/ardrivo/src/Wire.cxx

```diff
@@ -3,78 +3,66 @@
 #include "Arduino.h"
 #include "BoardDataDef.hxx"
 
-//Do diffrently??
-bool transmitting = false;
-uint8_t transmissionAdress;
-uint8_t transmitted;
+//What sets bus_id?? where does it update?
+//use protocolrequest to track writing to tx?
+//Fix the access with unordered map stuff, gives error atm
 
 void TwoWire::begin()
 {
     begun = true;
 }
 
-void TwoWire::begin(uint8_t adress)
+void TwoWire::begin(uint8_t address)
 {
-    //Begin to the specific "adress"???
+    slaveAddress = address;
     begun = true;
 }
-void TwoWire::begin(int adress)
+void TwoWire::begin(int address)
 {
-    //Begin to the specific "adress"???
-
+    slaveAddress = address;
     begun = true;
 }
 
+// requestFrom = REQUEST_SEND
 size_t TwoWire::requestFrom(uint8_t address, size_t size, bool stop)
 {
-    // ?????
-    // How do you diffrentiate REQUEST_SEND och REQUEST_RECEIVE????
-    /*    
-    REQUEST_SEND = 1,
-    REQUEST_RECEIVE
-    */
-
-    //vad faeeen g�r x??
-    //int x = 0 och x i ic2_buses[x] ????
-
-    //med denna gn�ller den inte
+ 
     ProtocolRequest req = ProtocolRequest(RequestType::REQUEST_SEND, size);
-    transmissionAdress = address;
+    transmissionAddress = address;
     
-    //med denna gn�ller den 
     //auto req = new ProtocolRequest(RequestType::REQUEST_SEND, size);
+
     board_data->i2c_buses[bus_id].requests.at(address).push_back(req);
 
-    // need other return value
-    return 0;
+    size_t bytesReturned = req.bytes;
+    return bytesReturned;
 }
 
 uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size)
 {
-    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+    return requestFrom(address, size, static_cast<uint8_t>(true));
 }
 uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size, uint8_t stop)
 {
-    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+    return requestFrom (address, size, static_cast<uint8_t>(true));
 }
 uint8_t TwoWire::requestFrom(int address, int size)
 {
-    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+    return requestFrom(static_cast<uint8_t>(true), static_cast<uint8_t>(true), static_cast<uint8_t>(true));
 }
 uint8_t TwoWire::requestFrom(int address, int size, int stop)
 {
-    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)stop);
-
+    return requestFrom(static_cast<uint8_t>(true), static_cast<uint8_t>(true),static_cast<uint8_t>(true));
```

Comment:
A function which take three parameters and you don't even use any of them; smells like a mistake even without reading the rest of the file